filestore/src/main/java/com/yahoo/ycsb/db/FileStoreClient.java

```diff
@@ -0,0 +1,233 @@
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.*;
+
+import static java.io.File.separatorChar;
+
+/**
+ * This "database" creates a file on the given path from properties for each insert call.
+ * Is should be used to ensure that the data is the same on multiple benchmark runs with different databases.
+ */
+public class FileStoreClient extends DB {
+
+  /**
+   * The name and default value of the property for the output directory for the files.
+   */
+  public static final String OUTPUT_DIRECTORY_PROPERTY = "outputDirectory";
+  public static final String OUTPUT_DIRECTORY_DEFAULT = System.getProperty("user.dir")
+      + separatorChar
+      + "benchmarkingData"
+      + separatorChar;
+
+  /**
+   * The property name to enable pretty printing of the json in created files.
+   * This will increase the size of the files substantially!
+   */
+  public static final String ENABLE_PRETTY_PRINTING = "enablePrettyPrinting";
+  public static final String ENABLE_PRETTY_PRINTING_DEFAULT = "false";
+
+  private final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeAdapter(ByteIterator.class, new
+      ByteIteratorAdapter());
+  private Gson gson;
+  private final Type valuesType = new TypeToken<Map<String, ByteIterator>>() {}.getType();
+
+  private String outputDirectory;
+
+  @Override
+  public void init() throws DBException {
+    Properties properties = getProperties();
+    outputDirectory = properties.getProperty(OUTPUT_DIRECTORY_PROPERTY, OUTPUT_DIRECTORY_DEFAULT);
+
+    if (outputDirectory.charAt(outputDirectory.length() - 1) != separatorChar) {
+      outputDirectory += separatorChar;
+    }
+
+    File directory = new File(outputDirectory);
+
+    if (!directory.exists() && !directory.mkdirs()) {
+      throw new DBException("Could not create output directory for files with path: " + outputDirectory);
+    }
+
+    if (!properties.getProperty(ENABLE_PRETTY_PRINTING, ENABLE_PRETTY_PRINTING_DEFAULT).equals("false")) {
+      gsonBuilder.setPrettyPrinting();
+    }
+
+    gson = gsonBuilder.create();
+  }
+
+  /**
+   * Reads the file with the name {@code table}_{@code key}.json.
```

Comment:
This layout is going to create directories that folks have a hard time managing with even modest datasets. if we must keep with one-file-per-key, then I'd suggest making directories within the data directory. Something like 3 or 4 layers based on a prefix of the key.